pyiceberg/view/metadata.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Dict, List, Literal, Optional
+
+from pydantic import Field, field_validator
+
+from pyiceberg.schema import Schema
+from pyiceberg.typedef import IcebergBaseModel, Properties
+from pyiceberg.types import transform_dict_value_to_str
+
+
+class ViewRepresentation(IcebergBaseModel):
+    """Represents the SQL query that defines the view."""
+
+    type: Literal["sql"] = Field()
+    """A string that indicates the type of representation. Must be `sql`"""
+    sql: str = Field()
+    """A string that contains the SQL text of the view definition."""
+    dialect: str = Field()
+    """The dialect of the SQL, e.g. `spark`, `trino`, `presto`."""
+
```

Comment:
this seems to be an implementation of the [SQLViewRepresentation](https://github.com/apache/iceberg/blob/83cc7d8749f2ba7ed7f0bff09e7bcf10d77785e8/api/src/main/java/org/apache/iceberg/view/SQLViewRepresentation.java) instead of the base [ViewRepresentation](https://github.com/apache/iceberg/blob/83cc7d8749f2ba7ed7f0bff09e7bcf10d77785e8/api/src/main/java/org/apache/iceberg/view/ViewRepresentation.java)

maybe this is fine but just wanted to call it out